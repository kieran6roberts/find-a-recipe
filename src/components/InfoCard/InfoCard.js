import React from "react";
import SubHeader from "../SubHeader/SubHeader";
import classes from "./InfoCard.module.css";

const InfoCard = ({ children, title, srcSet, sizes, src, imgclass, alt}) => {
  return(
    <div className={classes.flexItem}>
      <SubHeader>
        {title}
      </SubHeader>
      <div className={classes.info}>
        {children}
      </div>
      <img 
      srcSet={srcSet}
      sizes={sizes}
      src={src} 
      className={imgclass} 
      alt={alt} />
  </div>
  )
};

export default InfoCard;