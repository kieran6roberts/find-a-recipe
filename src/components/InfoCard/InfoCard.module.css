.info {
  padding: var(--space-md) 0;
  font-size: var(--text-sm);
}

.link {
  color: var(--primary-color);
  font-weight: 700;
}

@media only screen and (min-width: 700px) {
  .info {
    width: 75%;
  }
}

@media only screen and (min-width: 1800px) {
  .flexItem {
    flex: 1;
  }
  .flexItem > * {
    margin-bottom: var(--space-md);
  }
}