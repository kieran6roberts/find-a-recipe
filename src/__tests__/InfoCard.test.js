import InfoCard from "../components/InfoCard/InfoCard";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";


describe("<InfoCard />", () => {
    test("snapshot", () => {
      const tree = renderer.create(
        <InfoCard 
          title="title"
          srcSet="srcset"
          sizes="sizes"
          src="src"
          imgclass="class"
          alt="alt">
              Navigate to the search page to begin searching for recipes.
              Search by keywords that would included in the name of the recipe.
          </InfoCard>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    test("renders with correct props", () => {
      render(
        <InfoCard 
          title="title"
          srcSet="srcset"
          sizes="sizes"
          src="src"
          imgclass="class"
          alt="alt">
              Navigate to the search page to begin searching for recipes.
              Search by keywords that would included in the name of the recipe.
          </InfoCard>
      );
      expect(screen.getByText("title")).toBeInTheDocument();
      expect(screen.getByAltText("alt")).toBeInTheDocument();
      expect(screen.getByText(/Navigate to the search page/)).toBeInTheDocument();
    });
});