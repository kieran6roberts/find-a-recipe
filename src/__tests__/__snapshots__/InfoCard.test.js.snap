// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InfoCard /> snapshot 1`] = `
<div
  className="flexItem"
>
  <h2
    className="header"
  >
    title
  </h2>
  <div
    className="info"
  >
    Navigate to the search page to begin searching for recipes. Search by keywords that would included in the name of the recipe.
  </div>
  <img
    alt="alt"
    className="class"
    sizes="sizes"
    src="src"
    srcSet="srcset"
  />
</div>
`;
